File: src/main/java/itmo/is/project/mapper/resource/StoredResourceMapper.java
package itmo.is.project.mapper.resource;

import itmo.is.project.dto.resource.StoredResourceDto;
import itmo.is.project.mapper.EntityMapper;
import itmo.is.project.model.module.storage.StoredResource;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

@Mapper(componentModel = "spring")
public interface StoredResourceMapper extends EntityMapper<StoredResourceDto, StoredResource> {

    @Override
    @Mapping(source = "id.storageModuleId", target = "storageModuleId")
    StoredResourceDto toDto(StoredResource entity);
}

File: src/main/kotlin/itmo/isproject/mapper/module/storage/StoredResourceMapper.kt
package itmo.isproject.mapper.module.storage

import itmo.isproject.dto.module.storage.StoredResourceDto
import itmo.isproject.mapper.EntityMapper
import itmo.isproject.model.module.storage.StoredResource
import org.mapstruct.Mapper
import org.mapstruct.Mapping

@Mapper
interface StoredResourceMapper : EntityMapper<StoredResourceDto, StoredResource> {

    @Mapping(source = "storageModule.id", target = "storageModuleId")
    override fun toDto(entity: StoredResource): StoredResourceDto
}
